### DISCUSSION TITLE
User values from Query in Options Picker

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9094] [Help]
Hi there,

I've was to populate the values in an options picker with the values from a (MySQL) query. I've tried via Custom in the Options source, but I do not (yet ;-) see how I can do this.

Thanks in advance for your help!

**Comment by diogenesbrussels** [comment]
Hey @ecxs 

1. Create a MySQL query
2. Add a Data Provider
3. Select your query with this Data Provider
4. Add your picker inside the Data Provider
5. Select Data Provider as the Options Source
6. Select your Data Provider as the Options Provider

I hope it helps you!

**Reply by ecxs** [reply]
Hi @diogenesbrussels thanks, but if I understand correctly, this is when you want to populate the values in the options picker with the data from the field in the datasource. However, I would like to get this data from another datasource, being a separate valuelists table ...

**Reply by diogenesbrussels** [reply]
https://docs.budibase.com/docs/option-picker#using-a-data-provider

**Reply by diogenesbrussels** [reply]
@ecxs 
I'm not sure I understand, maybe add a second Data Provider with this other data source?

**Reply by ecxs** [reply]
@diogenesbrussels a second Data Provider does not work. The Data provider only seems to work on the table that the form is based on, ergo the value list is based on the (dynamic) data that is in the field. With the Custom Method your can define your own options (values). The online docs mention 'Using the custom method, you can also: - populate values in the form, from an API request, or custom query.' But, unfortunately, I can't get this to work, and can not find any further reading on this ...

See https://docs.budibase.com/docs/option-picker
